### PR TITLE
CI: Migrate to GitHub Actions

### PR DESCRIPTION
**Pull request by ichizok**
On Travis-CI run only linux s390x and arm64 tests.

### CODE REVIEW

**Comment by k-takata** [comment]
Link to the discussion in the vim_dev mailing list:
https://groups.google.com/g/vim_dev/c/fj7d7pBllx0

**Comment by brammool** [comment]
Looks like there are asan failures.  Perhaps EXITFREE is missing?

**Comment by ichizok** [comment]
> Looks like there are asan failures. Perhaps EXITFREE is missing?

As far as seeing the build log, `-DEXITFREE` was applied...
And, sorry, the result of linux-huge/clang (and a few jobs) is not stable yet.

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -179,10 +179,10 @@ jobs:
...
         if: matrix.coverage && success()
         run: |
-          sudo apt-get install -y python3-setuptools python3-wheel
+          #sudo apt-get install -y python3-setuptools python3-wheel
           # needed for https support for coveralls building cffi only works with gcc, not with clang
-          CC=gcc pip3 install --user cpp-coveralls pyopenssl ndg-httpsclient pyasn1
-          ~/.local/bin/coveralls -b "${SRCDIR}" -x .xs -e "${SRCDIR}"/if_perl.c -e "${SRCDIR}"/xxd -e "${SRCDIR}"/libvterm --encodings utf-8 -t "${{ secrets.GITHUB_TOKEN }}"
+          #CC=gcc pip3 install --user cpp-coveralls pyopenssl ndg-httpsclient pyasn1
+          #~/.local/bin/coveralls -b "${SRCDIR}" -x .xs -e "${SRCDIR}"/if_perl.c -e "${SRCDIR}"/xxd -e "${SRCDIR}"/libvterm --encodings utf-8 -t "${{ secrets.GITHUB_TOKEN }}"
           (cd "${SRCDIR}" && bash <(curl -s https://codecov.io/bash))
```
Comment (k-takata) [inline .github/workflows/ci.yml]: Is the `git` command available in the $PATH?
The codecov's bash script may not work correctly if git is not available.

Comment (k-takata) [reply .github/workflows/ci.yml]: Ah, sorry. It seems that I commented on an outdated commit.

**Comment by ichizok** [comment]
Another problem:
I haven't found snd-dummy.ko in GHA ubuntu environment. snd-dummy.ko is not included in linux-modules-extra-azure package.

**Comment by ichizok** [comment]
Coveralls needs`COVERALLS_REPO_TOKEN`, set it to the repository secrets.
(Repository's Settings > Secrets > New repository secrets)

**Comment by ichizok** [comment]
I committed "Improve StopVimInTerminal", because:

When build Vim with coverage option by clang-8 or newer, Test_popupwin_latin1_encoding often fails.
As far as I investigated, sometimes `fork()` (in `mch_job_start()`) is delayed and Vim in terminal cannot receive quit command sent just after starting.
I fixed it to send quit command several times.

**Comment by chrisbra** [comment]
> Coveralls needsCOVERALLS_REPO_TOKEN, set it to the repository secrets.
(Repository's Settings > Secrets > New repository secrets)

should be there now.

**Comment by ichizok** [comment]
> > Coveralls needsCOVERALLS_REPO_TOKEN, set it to the repository secrets.
> > (Repository's Settings > Secrets > New repository secrets)
> 
> should be there now.

Oh, it seems the outer repository cannot fetch this secrets, thus coveralls cannot succeed until merged.
This PR is ready to merge.

**Comment by brammool** [comment]
> I committed "Improve StopVimInTerminal", because:
> 
> When build Vim with coverage option by clang-8 or newer, Test_popupwin_latin1_encoding often fails.
> As far as I investigated: sometimes `fork()` (in `mch_job_start()`) is
> delayed and Vim in terminal cannot receive quit command sent just
> after starting.
> I fixed it to send quit command several times.

Looks like it only waits for Vim to startup, not for the script to
finish.  I'll add an echo command and wait for that text to show.

-- 
ARTHUR: Old woman!
DENNIS: Man!
ARTHUR: Man.  I'm sorry.  Old man, What knight live in that castle over there?
DENNIS: I'm thirty-seven.
                 "Monty Python and the Holy Grail" PYTHON (MONTY) PICTURES LTD

 /// Bram Moolenaar -- Bram@Moolenaar.net -- http://www.Moolenaar.net   \\\
///        sponsor Vim, vote for features -- http://www.Vim.org/sponsor/ \\\
\\\  an exciting new programming language -- http://www.Zimbu.org        ///
 \\\            help me help AIDS victims -- http://ICCF-Holland.org    ///

**Comment by ichizok** [comment]
> Looks like it only waits for Vim to startup, not for the script to
> finish.  I'll add an echo command and wait for that text to show.

Merged 8.2.2149, but Test_popupwin_latin1_encoding still failed.

https://github.com/vim/vim/pull/7433/checks?check_run_id=1572095335

**Comment by brammool** [comment]
> > Looks like it only waits for Vim to startup, not for the script to
> > finish.  I'll add an echo command and wait for that text to show.
> 
> Merged 8.2.2149, but Test_popupwin_latin1_encoding still failed.
> 
> https://github.com/vim/vim/pull/7433/checks?check_run_id=1572095335

Hmm, not clear why that happens.

-- 
Looking at Perl through Lisp glasses, Perl looks atrocious.

 /// Bram Moolenaar -- Bram@Moolenaar.net -- http://www.Moolenaar.net   \\\
///        sponsor Vim, vote for features -- http://www.Vim.org/sponsor/ \\\
\\\  an exciting new programming language -- http://www.Zimbu.org        ///
 \\\            help me help AIDS victims -- http://ICCF-Holland.org    ///

**Comment by brammool** [comment]
I see this warning: "1 issue was detected with this workflow: git checkout HEAD^2 is no longer necessary. Please remove this step as Code Scanning recommends analyzing the merge commit for best results."
Should we remove line 41 from codeql-analysis.yml ?

**Comment by chrisbra** [comment]
> Should we remove line 41 from codeql-analysis.yml ?

I suppose we can remove the whole block then, if codeql can analyze the merge commit now properly.

https://github.com/vim/vim/blob/0353f56ddb379e7f1a68172fa4743355e04df21e/.github/workflows/codeql-analysis.yml#L39-L42

**Pull request closed without merging**